File: src/Binary/BinarySearch.java
package Binary;

// https://leetcode.com/problems/binary-search/
public class BinarySearch {
    public int search(int[] nums, int target) {
        int i = 0;
        int j = nums.length - 1;

        while (i <= j) {
            int mid = i + (j - i)/2;
            if (nums[mid] > target) {
                j = mid - 1;
            } else if (nums[mid] < target) {
                i = mid + 1;
            } else {
                return mid;
            }
        }
        return -1;
    }

    public static void main(String[] args) {
        BinarySearch b = new BinarySearch();
        int[] input = {-1,0,3,5,9,12};
        System.out.println(b.search(input, -1));
    }
}

File: py/essential/BinarySearch.py
# https://leetcode.com/problems/binary-search/description/
# Grind
# Neet

from typing import List


class Solution:
    def search(self, nums: List[int], target: int) -> int:
        left, right = 0, len(nums) - 1

        while left <= right:
            mid = (left + right) // 2
            if nums[mid] == target:
                return mid
            if nums[mid] > target:
                right = mid - 1
            else:
                left = mid + 1
        return -1


s = Solution()
nums = [2, 5]
target = 5
print(s.search(nums, target))
